### ISSUE TITLE
Batching does not work when concat op is present in the graph

### ISSUE THREAD

**Issue by kikushah** [tensorflow/serving#1304]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow-serving

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with
   documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues.
We want to focus on work that benefits the whole community, e.g., fixing bugs
and adding features. Support only helps individuals. GitHub also notifies
thousands of people when issues are filed. We want them to see you communicating
an interesting problem, rather than being redirected to Stack Overflow.

-----------------------
## Feature Request
If this is a feature request, please fill out the following form in full:

### Describe the problem the feature is intended to solve
A clear and concise description of what the problem is. Ex. I'm always
frustrated when [...]

### Describe alternatives you've considered
A clear and concise description of any alternative solutions or features you've
considered.

## Bug Report
If this is a bug report, please fill out the following form in full:

### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow Serving installed from (source or binary)**:
- **TensorFlow Serving version**:

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in
TensorFlow Serving.

### Exact Steps to Reproduce
Please include all steps necessary for someone to reproduce this issue on their
own machine. If not, skip this section.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem.
If including tracebacks, please include the full traceback. Large logs and files
should be attached. Try to provide a reproducible test case that is the bare
minimum necessary to generate the problem.

**Comment by kikushah** [comment]
Hi,

I trained a model using python 3.6 and tensorflow 1.13.

I am training a densenet. One of the ops that I use is concat(with tf.zeros), to double the size of my input tensor. The model is trained with batch size 1.

It trains well, and the graph is exported with tf.saved_model.builder.SavedModelBuilder and served with tensorflow_model_server.

I use a c++ client to send prediction requests to the served model, from localhost.

When I send requests of batch size 1, it works well.

When I enable batching and increase the max batch size to a value greater than 1, the concat op fails. 

Upon debugging, I found that the requests are getting batched properly but there is a problem in the concat op.

`gRPC call return code: 3: _MklConcatOp : Dimensions of inputs should match: shape[0][0]= 10 vs. shape[1][0] = 1
         [[{{node global_Model/rn_first/concat}} = _MklConcatV2[N=2, T=DT_FLOAT, Tidx=DT_INT32, _kernel="MklOp", _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_global_Model/global_Model/PlaceholderWithDefault_0_0, global_Model/global_Model/PlaceholderWithDefault/input, global_Model_-1/ArgMax/dimension, DMT/_0, DMT/_1, DMT/_2)]]`

While sending single requests, input tensor is of shape (1, 50). While sending synchronous batch requests, the input tensor shape is (10, 50). While sending asynchronous request, the shape is (1, 50) but 10 such requests.

I am not sure if I am doing anything wrong here. This seems to be a problem or bug in the serving batching.
